src/main/java/br/edu/ufcg/computacao/alumni/core/Match.java

```diff
@@ -0,0 +1,122 @@
+			return Curso.MESTRADO_EM_INFOMATICA;
+		}		
+		if (field.contains("") && (degree.contains("BC") || degree.contains("BACHELOR") || degree.contains("BS"))) {
+			return Curso.GRADUACAO_PROCESSAMENTO_DE_DADOS;
+		}
+		return null;
+	}
+	
+	private int getScoreFromName(String alumniName, String linkedinName) {
+		if (alumniName.equals(linkedinName)) return 1;
+		return 0;
+	}
+	
+	private int getScoreFromCurso(List<Curso> alumniCursos, Curso linkedinCurso) {
+		if (linkedinCurso == null) return 0;
+		if (alumniCursos.contains(linkedinCurso)) return 1;
+		return 0;
+	}
+	
+	private int getScoreFromSchoolUrl(String schoolUrl, String linkedinUrl) {
+		if (linkedinUrl == null) return 0;
+		if (schoolUrl.equals(linkedinUrl)) return 1;
+		return 0;
+	}
+	
+	private int getScoreFromYear(List<String> alumniYears, String linkedinYear) {
+		if (linkedinYear == null) return 0;
+		if (alumniYears.contains(linkedinYear)) return 1;
+		return 0;
+	}
+	
+	private <E> List<E> filterGraus(Grau[] graus, Function<Grau, E> f) {
+		return Arrays.asList(graus)
+				.stream()
+				.map(f)
+				.collect(Collectors.toList());
+	}
+	
+	public Collection<LinkedinAlumnusData> getMatches(UfcgAlumnusData alumni, String schoolUrl) {
+		try {
+			Collection<LinkedinAlumnusData> linkedinProfilesList = LinkedinDataHolder.getInstance().getLinkedinAlumniData();
+			Collection<LinkedinAlumnusData> selectedProfilesList = new LinkedList<>();
+			
+			String alumniName = alumni.getFullName();
+			Grau[] alumniGraus = alumni.getGraus();
+			
+			List<String> alumniStartYears = filterGraus(alumniGraus, Grau::getSemestreIngresso);
+			List<String> alumniEndYears = filterGraus(alumniGraus, Grau::getSemestreFormatura);
+			List<Curso> alumniCursos = filterGraus(alumniGraus, Grau::getCurso);
+			
+			linkedinProfilesList.forEach(linkedinProfile -> {
+				int score = 0;
+				
+				LinkedinSchoolData[] schoolData = linkedinProfile.getSchools();
+				String linkedinAlumniFullName = linkedinProfile.getFullName().toUpperCase();
+				
+				score += getScoreFromName(alumniName, linkedinAlumniFullName);
+				
+				for (LinkedinSchoolData school : schoolData) {
+					String linkedinUrl = school.getSchoolUrl();
+					String linkedinDegree = school.getDegree();
+					String linkedinField = school.getField();
+					String linkedinStartYear = school.getDateRange().getStartYear();
+					String linkedinEndYear = school.getDateRange().getEndYear();
+					Curso linkedinCurso = getCurso(linkedinField, linkedinDegree);
+					
+					score += getScoreFromCurso(alumniCursos, linkedinCurso);
+					score += getScoreFromYear(alumniEndYears, linkedinEndYear);
+					score += getScoreFromYear(alumniStartYears, linkedinStartYear);
+					score += getScoreFromSchoolUrl(schoolUrl, linkedinUrl);
+				}
+				
+				if (score >= 1) {
+					selectedProfilesList.add(linkedinProfile);
+				}
+			});
+			return selectedProfilesList;
+			
+		} catch (Exception e) {
```

Comment:
evite esse try/catch gigante. isso vai ser fonte de problemas no futuro. i.e. o método retornar nada, e vc sem entender pq.

só usa try/catch se for estritamente, garantidamente, necessário.

Reply:
era realmente necessário mesmo, pq quando eu chamo o método que retorna a coleção de perfis do linkedin ele reclama ao compilador e pede pra tratar com um throws ou um try/catch. aí eu optei pelo try/catch mesmo

Reply:
nesse caso, é melhor repassar o throws para deixar que a interface trate o erro. se realmente achar que o mais adequado é um try/catch. limita ele só pra linha 78... e mesmo assim, só capturando o erro específico que queres tratar.